.github/workflows/semantic-commits.yml

```
@@ -0,0 +1,21 @@
+name: "Check Semantic Commit"
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+jobs:
+  main:
+    name: Validate PR Title
+    runs-on: ubuntu-latest
+    steps:
+      - name: semantic-pull-request
+        uses: amannn/action-semantic-pull-request@91682d0
```

```suggestion
        uses: amannn/action-semantic-pull-request@@v4
```
Why not a version instead?

---

It's considered [a security best practice by GitHub](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) when using third-party actions:

> Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps mitigate the risk of a bad actor adding a backdoor to the action's repository, as they would need to generate a SHA-1 collision for a valid Git object payload.

Prevents a future supply chain attack if someone were to compromise that repo and change the tag to point at malicious code.

Added benefit of also ensuring the check doesn't break without us doing anything if the author inadvertently ships a buggy version.